python-frontend/src/main/java/org/sonar/python/semantic/FunctionSymbolImpl.java

```diff
@@ -352,6 +373,21 @@ public void setIsDjangoView(boolean isDjangoView) {
       this.annotatedTypeName = annotatedTypeName;
     }
 
+    public ParameterImpl(FunctionDescriptor.Parameter parameterDescriptor, ProjectLevelSymbolTable projectLevelSymbolTable) {
+      this.name = parameterDescriptor.name();
+      this.hasDefaultValue = parameterDescriptor.hasDefaultValue();
+      this.isVariadic = parameterDescriptor.isVariadic();
+      this.isKeywordOnly = parameterDescriptor.isKeywordOnly();
+      this.isPositionalOnly = parameterDescriptor.isPositionalOnly();
+      this.location = parameterDescriptor.location();
+      this.annotatedTypeName = parameterDescriptor.annotatedType();
+      Symbol typeSymbol = projectLevelSymbolTable.getSymbol(parameterDescriptor.annotatedType());
+      if (typeSymbol == null && annotatedTypeName != null) {
+        typeSymbol = symbolWithFQN(annotatedTypeName);
+      }
+      this.declaredType = typeSymbol == null ? anyType() : new DeclaredType(typeSymbol, Collections.emptyList());
```

Comment:
Good catch! Thanks!
I think it's rather important to fix this soon, but I agree it might be good to get this PR out of the way and do the fix in a separate one. I created SONARPY-951 and committed it to the current sprint.